invoice/src/main/java/org/killbill/billing/invoice/dao/CBADao.java

```diff
@@ -88,26 +85,27 @@ public InvoiceItemModelDao computeCBAComplexity(final InvoiceModelDao invoice,
         }
     }
 
-    private BigDecimal getInvoiceBalance(final InvoiceModelDao invoice) {
+    @VisibleForTesting
+    boolean isParentExistAndRawBalanceIsZero(final InvoiceModelDao invoice) {
+        return invoice.getParentInvoice() != null &&
+               InvoiceModelDaoHelper.getRawBalanceForRegularInvoice(invoice.getParentInvoice()).compareTo(BigDecimal.ZERO) == 0;
+    }
 
-        final InvoiceModelDao parentInvoice = invoice.getParentInvoice();
-        if ((parentInvoice != null) && (InvoiceModelDaoHelper.getRawBalanceForRegularInvoice(parentInvoice).compareTo(BigDecimal.ZERO) == 0)) {
-            final Iterable<InvoiceItemModelDao> items = Iterables.filter(parentInvoice.getInvoiceItems(), new Predicate<InvoiceItemModelDao>() {
-                @Override
-                public boolean apply(final InvoiceItemModelDao input) {
-                    return input.getChildAccountId().equals(invoice.getAccountId());
-                }
-            });
+    @VisibleForTesting
+    BigDecimal getChildInvoiceAmountCharged(final InvoiceModelDao invoice) {
+        return InvoiceModelDaoHelper.getAmountCharged(invoice);
+    }
 
-            final BigDecimal childInvoiceAmountCharged = InvoiceModelDaoHelper.getAmountCharged(invoice);
-            BigDecimal parentInvoiceAmountChargedForChild = BigDecimal.ZERO;
+    @VisibleForTesting
+    BigDecimal getInvoiceBalance(final InvoiceModelDao invoice) {
+        if (isParentExistAndRawBalanceIsZero(invoice)) {
+            final BigDecimal parentInvoiceAmountChargedForChild = invoice.getParentInvoice().getInvoiceItems().stream()
```

Comment:
⚠️ We used to have a filtering step `input.getChildAccountId().equals(invoice.getAccountId()`, and I am not seeing it anymore, is this an issue?

Reply:
You're right, it is get removed. I'll add it back and add additional test in `TestCBADaoUnit`

Reply:
👍 Please add a pointer to the fix, so we can review that part.

Reply:
[This](https://github.com/killbill/killbill/blob/6220755999c3be2ba4683ac4ec07fea1c9b86147/invoice/src/main/java/org/killbill/billing/invoice/dao/CBADao.java#L102) is the latest. [New test added ](https://github.com/killbill/killbill/blob/6220755999c3be2ba4683ac4ec07fea1c9b86147/invoice/src/test/java/org/killbill/billing/invoice/dao/TestCBADaoUnit.java#L94)for filtering step you mentioned above.

Is that what you ask?